src/main/java/net/atos/entng/rbs/filters/TypeAndResourceAppendPolicy.java

```diff
@@ -155,6 +155,108 @@ public void authorize(final HttpServerRequest request, final Binding binding, fi
+			// Query
+			StringBuilder query = new StringBuilder();
+			JsonArray values = new fr.wseduc.webutils.collections.JsonArray();
+			query.append("SELECT count(*)")
+					.append(" FROM rbs.resource AS r")
+					.append(" INNER JOIN rbs.resource_type AS t ON r.type_id = t.id");
+			if (hasBooking) {
+				// Additional join when parameter bookingId is used
+				query.append(" INNER JOIN rbs.booking AS b ON b.resource_id = r.id");
+			}
+
+			query.append(" LEFT JOIN rbs.resource_type_shares AS ts ON t.id = ts.resource_id")
+					.append(" LEFT JOIN rbs.resource_shares AS rs ON r.id = rs.resource_id")
+					.append(" WHERE (");
+
+			if (isUpdateBooking(binding) || isUpdatePeriodicBooking(binding) || isDeleteBooking(binding)) {
+				query.append("(ts.member_id IN ").append(Sql.listPrepared(groupsAndUserIds.toArray())).append(" AND ts.action = 'net-atos-entng-rbs-controllers-BookingController|processBooking')");
+				for (String groupOruser : groupsAndUserIds) {
+					values.add(groupOruser);
+				}
+
+				query.append(" OR (rs.member_id IN ").append(Sql.listPrepared(groupsAndUserIds.toArray())).append(" AND rs.action = 'net-atos-entng-rbs-controllers-BookingController|processBooking')");
+				for (String groupOruser : groupsAndUserIds) {
+					values.add(groupOruser);
+				}
+			} else {
+				query.append("(ts.member_id IN ").append(Sql.listPrepared(groupsAndUserIds.toArray())).append(" AND ts.action = ?)");
+				for (String groupOruser : groupsAndUserIds) {
+					values.add(groupOruser);
+				}
+				values.add(sharedMethod);
+
+				query.append(" OR (rs.member_id IN ").append(Sql.listPrepared(groupsAndUserIds.toArray())).append(" AND rs.action = ?)");
+				for (String groupOruser : groupsAndUserIds) {
+					values.add(groupOruser);
+				}
+				values.add(sharedMethod);
+			}
+
+			// Authorize user if he is a local administrator for the resourceType's school_id
+			List<String> scope = getLocalAdminScope(user);
+			if (scope!=null && !scope.isEmpty()) {
+				query.append(" OR t.school_id IN ").append(Sql.listPrepared(scope.toArray()));
+				for (String schoolId : scope) {
+					values.add(schoolId);
+				}
+			}
+
+			query.append(" OR t.owner = ? OR r.owner = ?");
+			values.add(user.getUserId()).add(user.getUserId());
+			if(isDeleteBooking(binding)) {
+				query.append(" OR b.owner = ?"); // Owner can delete his booking
+				values.add(user.getUserId());
+			} else if(isUpdateBooking(binding) || isUpdatePeriodicBooking(binding)) {
+				// Check that the user is the booking's owner
+				query.append(" OR b.owner = ?"); // Owner can delete his booking
+				values.add(user.getUserId());
+			}
+
+			query.append(") AND r.id = ?");
+			values.add(Sql.parseId(resourceId));
+			if (hasBooking) {
+				query.append(" AND b.id = ?");
+				values.add(Sql.parseId(bookingId));
+			}
+
+			if (isCreatePeriodicBooking(binding)) {
+				// Check that the resource allows periodic_booking
+				query.append(" AND r.periodic_booking = ?");
+				values.add(true);
```

Comment:
query.append(" AND r.periodic_booking = true"); directement ?

Reply:
authorize duplicate with arguments so that no HTTPRequest is needed (polymorph)